pkg/controllers/nodeclass/suite_test.go

```diff
@@ -107,6 +108,46 @@ var _ = Describe("NodeClassController", func() {
 			},
 		})
 	})
+	Context("Cluster CIDR Resolution", func() {
+		BeforeEach(func() {
+			// Cluster CIDR will only be resolved once per lifetime of the launch template provider, reset to nil between tests
+			awsEnv.LaunchTemplateProvider.ClusterCIDR = nil
+		})
+		It("should only resolve cluster CIDR for AL2023", func() {
```

Comment:
```suggestion
		It("shouldn't resolve cluster CIDR for non-Al2023 NodeClasses", func() {
```

nit: Since this isn't actually testing that this is resolving the AL2023 CIDR, consider being more explicit about the fact that this is testing that you are validating that you don't resolve the CIDR on non-AL2023 AMIFamilies